ML-Frameworks/utils/git-utils.sh

```diff
@@ -45,23 +45,42 @@ function apply-github-patch {
     # Apply a specific GitHub commit.
     # $1 is 'organisation/repo', $2 is the commit hash
     # To use an API token, which may avoid rate limits, set the environment variable GITHUB_TOKEN
-
     set -u
+
+    local github_url='https://github.com'
+    local github_api_url='https://api.github.com/repos'
+
     patch_file="$patch_cache_dir/$2.patch"
     if [ ! -f "$patch_file" ]; then
-        local github_api_url='https://api.github.com/repos'
-        local github_url='https://github.com'
-
         # Download the .patch file.
         if [[ "${GITHUB_TOKEN+x}" ]]; then
-            curl --silent -H "Authorization: token $GITHUB_TOKEN" -H "Accept: application/vnd.github.v3.patch" -L $github_api_url/$1/commits/$2 -o "$patch_file"
+            curl --silent \
+                -H "Authorization: token $GITHUB_TOKEN" -H "Accept: application/vnd.github.v3.patch" -L $github_api_url/$1/commits/$2 -o "$patch_file"
         else
             curl --silent -L $github_url/$1/commit/$2.patch -o "$patch_file"
         fi
     fi
 
-    # Apply the patch
+    # Three-way merge is less brittle than plain 'git am'. It can handle scenarios in which
+    # the context of the patch has moved. However, we need the parent commit of the patch for
```

Comment:
Very clever! Although it looks a bit bonkers. Do we still need the curl logic if we are going to use fetch? It feels a bit disjointed. Can we just fetch and cherry pick and acheive the same effect more simply?